vessel-runtime/src/main/java/com/textnow/android/vessel/VesselCacheImpl.kt

```diff
@@ -18,8 +18,12 @@ open class DefaultCache : VesselCache {
         return hashMap[key] as? T?
     }
 
-    override fun <T : Any> set(key: String, value: T) {
-        hashMap[key] = value
+    override fun <T : Any> set(key: String, value: T, fromPreload: Boolean) {
+        if (fromPreload && hashMap.containsKey(key)) {
```

Comment:
I'm just thinking through if there's any race condition here, where a cache entry could be updated out of order.  Not saying there is, just letting my brain ponder this in the background.

Reply:
The idea behind this is to allow preloading to happen asynchronously. Which has the potential to allow vessel writes to happen at the same time preloading is running. This change is meant to address that case.

Reply:
Revisiting this, there is a potential race condition between this check (being false) and writing the value.

Another (new) write done on a separate thread, in between the check and set, could result in a new write being over-written by a preloaded value.

To be fair this type of race condition already exists elsewhere in VesselImpl, but it wouldn't hurt to put a mutex/sycnronized block to ensure the check-and-set are atomic.